pkg/patterns/declarative/reconciler.go

```diff
@@ -359,10 +364,16 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 		}
 	}
 
+	// TODO(yuwenma): instead of passing in the applyset "parent" as a applier option, maybe it's better to pass in the
```

Comment:
I think to keep controller-runtime happy you do want to be working with the same instance of the subject.  It writes back the changed resourceversion I believe, so you can call update twice.